src/components/NavigationBar/SearchInput.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+	import { slide } from 'svelte/transition';
+	import { enhance } from '$app/forms';
+	import SearchIcon from 'carbon-icons-svelte/lib/Search.svelte';
+	import CloseIcon from 'carbon-icons-svelte/lib/Close.svelte';
+	import { createEventDispatcher } from 'svelte';
+
+	const dispatch = createEventDispatcher();
+
+	export let searchActionLink = '';
+	export let activeSearch = false;
+	let searchButton: HTMLButtonElement;
+	let searchInput: HTMLInputElement;
+	let searchForm: HTMLFormElement;
+
+	function searchClickHandle() {
+		if (!activeSearch) {
+			activeSearch = true;
+			dispatch('activate');
+		}
+	}
+
+	function introEndHandler() {
+		searchInput.focus();
+		searchButton.type = 'submit';
+	}
+
+	function closeClickHandle() {
+		activeSearch = false;
+		searchInput.value = '';
+		searchButton.type = 'button';
```

Comment:
Instead of accessing element property directly like `element.value` and `.type`, Svelte recommends a more declarative way which is binding the variable to the property in the template like
```svelte
<input bind:value />
<button type={buttonType} />
```
It will make the code easier to understand and debug.